Content.Goobstation.Shared/MartialArts/SharedMartialArtsSystem.SpaceBear.cs

```diff
@@ -0,0 +1,125 @@
+using Content.Goobstation.Common.MartialArts;
+using Content.Goobstation.Shared.MartialArts.Components;
+using Content.Goobstation.Shared.MartialArts.Events;
+using Content.Goobstation.Shared.Stunnable;
+using Content.Goobstation.Shared.Temperature;
+using Content.Goobstation.Shared.Temperature.Components;
+using Content.Shared._Shitmed.Targeting;
+using Content.Shared.Atmos.Components;
+using Content.Shared.Body.Systems;
+using Content.Shared.Clothing.Components;
+using Content.Shared.Clothing.EntitySystems;
+using Content.Shared.Damage;
+using Content.Shared.Examine;
+using Content.Shared.IdentityManagement;
+using Content.Shared.Interaction.Events;
+
+namespace Content.Goobstation.Shared.MartialArts;
+
+public partial class SharedMartialArtsSystem
+{
+    private void InitializeSpaceBear()
+    {
+        SubscribeLocalEvent<CanPerformComboComponent, BearJawsPerformedEvent>(OnBearJaws);
+        SubscribeLocalEvent<CanPerformComboComponent, PawSlamPerformedEvent>(OnPawSlam);
+        SubscribeLocalEvent<CanPerformComboComponent, BearSmokeyPerformedEvent>(OnBearSmokey);
+
+        SubscribeLocalEvent<GrantSpaceBearComponent, UseInHandEvent>(OnGrantCQCUse);
+        SubscribeLocalEvent<GrantSpaceBearComponent, ExaminedEvent>(OnGrantCQCExamine);
+
+    }
+
+    #region Generic Methods
+
+    private void OnGrantCQCUse(EntityUid ent, GrantSpaceBearComponent comp, UseInHandEvent args)
+    {
+        if (!_netManager.IsServer)
+            return;
+
+        if (comp.Used)
+        {
+            _popupSystem.PopupEntity(Loc.GetString(comp.LearnFailMessage, ("manual", Identity.Entity(ent, EntityManager))),
+                args.User,
+                args.User);
+            return;
+        }
+
+        if (!TryGrantMartialArt(args.User, comp))
+            return;
+        _popupSystem.PopupEntity(Loc.GetString(comp.LearnMessage), args.User, args.User);
+        comp.Used = true;
+        var slots = _bodySystem.GetBodyChildren(args.User);
+        foreach (var (slot,_) in slots)
+        {
+            EnsureComp<SpecialHighTempImmunityComponent>(slot);
+            _damageable.SetDamageModifierSetId(slot, "HeatImmunity");
+        }
+        EnsureComp<SpecialHighTempImmunityComponent>(args.User);
+        _faction.AddFaction(args.User, "SimpleHostile");
+        _damageable.SetDamageModifierSetId(args.User, "HeatImmunity");
```

Comment:
Modifying body parts is probably fine but you shouldn't modify the entity itself. Subscribe to DamageModifyEvent for damage modifier and... SpecialHighTempImmunity isn't used anywhere else so it's fine to leave it as is as well probably.
So just subscribe MartialArtKnowledgeComponent to DamageModifyEvent and apply modifier there. But there is a chance that it is already subscribed to in dragon kung fu system, I don't remember. In this case move it to base MartialArtsSystem, and make a switch statement for martial art knowledge. If it's KungFuDragon, then call method in dragon kung fu system, if it is space bear, then call method in space bear system.
The problem with current approach is that species lose their current species damage modifiers when learning this martial art.